### PR TITLE
Support filter by domain for observation APIs

### PR DESCRIPTION
**Pull request by shifucun**

### CODE REVIEW

File: internal/server/v2/observation/contained_in.go [hunk 1]
```diff
@@ -120,6 +122,15 @@ func FetchContainedIn(
 				sort.Sort(ranking.CohortByRank(cohorts))
 				for _, cohort := range cohorts {
 					facet := util.GetFacet(cohort)
+					if filter != nil && filter.Domain != "" {
+						url, err := url.Parse(facet.ProvenanceUrl)
+						if err != nil {
+							return nil, err
+						}
+						if !strings.HasSuffix(url.Hostname(), filter.Domain) {
```
Comment (pradh) [inline internal/server/v2/observation/contained_in.go:130]: Add a line of comment for the matching logic?

**Review by pradh — APPROVED** [review]
Nice!

File: internal/server/v2/observation/contained_in.go (hunk 1 above)
Comment (shifucun) [reply internal/server/v2/observation/contained_in.go:132]: Done

**Review by shifucun — COMMENTED** [review]
thanks!

**Pull request merged**